### ISSUE TITLE
Deal with plurality in resource names

### ISSUE THREAD

**Issue by matthchr** [Azure/k8s-infra#141]
For example, in `batch_accounts_types_gen.go` the type currently generated is 
```
// +kubebuilder:object:root=true
type BatchAccounts struct {
    ....
}
```

It has a `BatchAccountsSpec`, etc -- these should not be plural.

You can see a similar problem on the VMSS resources as well, for example: `virtual_machine_scale_sets_types_gen.go` has `VirtualMachineScaleSetsSpec` which should probably be `VirtualMachineScaleSetSpec` like it is in the handcrafted code

**Comment by theunrepentantgeek** [comment]
I don't think these can be handled automagically - just removing a trailing **s** is likely to go wrong. But, this is what the translations in `IdentifierFactory` are intended to handle.

**Comment by theunrepentantgeek** [comment]
Though, I've just had a look and it's not quite up to the task. I've a pretty good idea of what's needed to make this work though.

**Comment by matthchr** [comment]
@theunrepentantgeek should we close this now?

**Comment by theunrepentantgeek** [comment]
Yes; I thought saying "Resolves #141" on the other issue would close this one automagically.
